core/src/ModelArray.cpp

```diff
@@ -276,6 +274,40 @@ ModelArray::MultiDim ModelArray::locationFromIndex(Type type, size_t index)
     return loc;
 }
 
+void ModelArray::checkLimits(const ModelArray& mask) const
+{
+    // Mask the data with the land mask
+    const auto masked = (mask.data() == 1).select(m_data.col(0), fillValue);
+
+    // Check first for NaNs. The code is different for the bounds check, because Eigen doesn't
+    // return an index for NaN-checking.
+    if (masked.isNaN().any())
+        throw std::runtime_error("Field contains NaN.");
+
+    // Now we check the bounds and set the array index (i) and value if we're out of bounds
+    size_t i;
+    double value;
+    if (masked.minCoeff() < lowerPhysicalLimit) {
+        value = masked.col(0).minCoeff(&i);
+    } else if (masked.maxCoeff() > upperPhysicalLimit) {
+        value = masked.col(0).maxCoeff(&i);
```

Comment:
I would suggest changing the limits to a `<=` test. For example, if one were to apply strict limits to ice concentration, then the allowable values would be 0≤c_ice≤1, with both 0 and 1 as allowed values. To set this limit with the current implementation would require setting limits of `{-std::numeric_limits<double>::min(), 1.0+std::numeric_limits<double>::epsilon()}`. This has the knock-on effect of requiring the default limits to be `{-std::numeric_limits<double>::max(), std::numeric_limits<double>::max()}`, rather than using `std::numeric_limits<double>::infinity()`.

Reply:
No, but I'll concede that the code is not very intuitive.

The check returns true if the value is _outside_ the allowed range, in which case we ask Eigen to find the offending value and its location and proceed to throw an error.

So, with the current code, an initialisation of cice (for example) would be

```
cice(ModelArray::Type::DG, {0, 1})
```

This gives an allowed range of ``0 ≤ cice ≤ 1``, because as long as this is true, neither the ``if`` on line 290 nor the ``else if`` on line 292 is triggered, which brings us to a ``return`` on line 303.